pkg/kubelet/kubelet.go

```diff
@@ -110,8 +110,8 @@ const (
 	// Max amount of time to wait for the container runtime to come up.
 	maxWaitForContainerRuntime = 30 * time.Second
 
-	// nodeStatusUpdateRetry specifies how many times kubelet retries when posting node status failed.
-	nodeStatusUpdateRetry = 5
+	// nodeStatusUpdateRetryBackoff is the amount of time that we are willing to spend retrying to connect to the apiserver.
+	nodeStatusUpdateRetryMaximumTime = 20 * time.Minute
```

Comment:
Node Controller today defines maximumGracePeriod (5min) at:
https://github.com/kubernetes/kubernetes/blob/4e3bbe3915832b8f7d273014bb6b369c06df9186/pkg/controller/node/nodecontroller.go#L248

Once a given node doesn't update its status for 5 mins, all pods on the node will be evicted.